bundles/org.openhab.binding.magentatv/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,238 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="magentatv"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="receiver">
+		<label>MagentaTV Media Receiver</label>
+		<description>Represents a Telekom Media Receiver for MagentaTV</description>
+
+		<channel-groups>
+			<channel-group id="control" typeId="control">
+				<label>Control</label>
+			</channel-group>
+			<channel-group id="program" typeId="program">
+				<label>Program Information</label>
+			</channel-group>
+			<channel-group id="status" typeId="status">
+				<label>Play Status</label>
+			</channel-group>
+		</channel-groups>
+
+
+		<config-description>
+			<parameter name="udn" type="text">
+				<label>Unique Device Name</label>
+				<description>The UDN identifies the Media Receiver</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="modelId" type="text">
+				<label>Receiver Model</label>
+				<description>Model type (MR4xx/ MR2xx)</description>
+				<required>true</required>
+				<options>
+					<option value="Auto">AUTO</option>
+					<option value="DMS_TPB">MR400</option>
+					<option value="MR401B">MR401B</option>
+					<option value="MR201">MR201</option>
+				</options>
+			</parameter>
+			<parameter name="ipAddress" type="text">
+				<label>Media Receiver's IP Address</label>
+				<description>IP Address of the receiver</description>
+				<required>true</required>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="port" type="text">
+				<label>Receiver Port</label>
+				<description>Port address for UPnP</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="accountName" type="text">
+				<label>Account Name</label>
+				<description>Credentials: Login name (xxx@t-online.de, same as for the Telekom Kundencenter)</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="accountPassword" type="text">
+				<label>Account Password</label>
+				<description>Credentials: Account Password (same as for the Telekom Kundencenter)</description>
+				<context>password</context>
+				<required>true</required>
+			</parameter>
+			<parameter name="userId" type="text">
+				<label>User ID</label>
+				<description>Technical User ID for pairing</description>
+				<required>false</required>
```

Comment:
Well, this one IS required for the Thing handler to work, isn't it?

Reply:
Nope, default are the credentials, which are used to auto-load the userID

Reply:
No, credentials should not be provided in a textual configuration. And what the handler needs to work is the userId, nothing else.

Reply:
see above, various other bindings have user/password information in the thing config, most of them don't even remove them after successful login, no user complained so far (even I understand where you are coming from)

Reply:
See above, in all situations we end up with the Thing having the UID configured and not being able to work before this is the case - hence it is a required parameter.

Reply:
Refactired the whole topic, userId could now be generated by
- OH console command
- entering credentials in thing config

Therefore it's no longer mandantory